lib/netext/dialer.go

```diff
@@ -149,6 +141,66 @@ func (d *Dialer) GetTrail(
 	}
 }
 
+func (d *Dialer) dialAddr(addr string) (string, error) {
+	remote, err := d.findRemote(addr)
+	if err != nil {
+		return "", err
+	}
+
+	for _, ipnet := range d.Blacklist {
+		if (*net.IPNet)(ipnet).Contains(remote.IP) {
```

Comment:
I find this cast ... strange ... maybe we should change `lib.IPNet` so it isn't required
cc @na-- @imiric